src/cmdstan/command.hpp

```diff
@@ -219,7 +223,7 @@ int command(int argc, const char *argv[]) {
 
   // Instantiate model
   stan::model::model_base &model
-      = new_model(*var_context, random_seed, &std::cout);
+      = new_model(*var_context, random_seed, &std::cerr);
```

Comment:
Will this make `print` statements in the Stan model go to stderr?

Reply:
excellent question.  will check.

Reply:
answer is no.  the generated model.hpp file contains the function `new_model`; this instantiates the model object.

the instantiated model object is passed into the services methods along with stderr and stdout, and therefore the correct output streams are hooked up to the calls to `log_prob` et al.

can't connect the c++ dots to understand why calls to `rethrow_located` in the model constructor go to stderr, but they do.

Reply:
`print` statements in `transformed data` block will go to std err, because those are rolled into the model constructor.  everything else goes to stdout.

Reply:
also functions that print stuff will print to stderr when called in transformed data block, otherwise, will print to stdout.

Reply:
I do not think this should be moved to cerr as it will just be more confusing. 

As far as I understand, this stream for the constructor is not used anywhere apart from prints in transformed data. model instantiation failures are caught by https://github.com/stan-dev/stan/blob/develop/src/stan/io/validate_dims.hpp#L42

Log prob is passed the logger (https://github.com/stan-dev/cmdstan/blob/develop/src/cmdstan/command.hpp#L104).

If we need to change something in the codegen to make a proper solution for all of this or any larger refactor, lets do that. I  am against a quick fix that would also print users debug messages to stderr.

Reply:
Sounds like this is a cleanup for stanc3. Since it's only transformed data prints, then I think revert this change and just do the return code cleanup here.

Reply:
the model instantiation failures caught by `validate_dims` are printed to whatever output stream is passed in to `new_model`.    I spent an hour yesterday trying to figure out why - I don't see an output stream passed around, and yet, that's where these error messages end up.

the result it that it's necessary to parse stdout to figure out if the data supplied to model is OK.  cf https://github.com/stan-dev/cmdstanpy/blob/2a7a9a5cc3a8e37670092c0af11d1e4c6d252381/cmdstanpy/stanfit.py#L230-L244. - this is a quick and dirty parse - it's brittle, might catch too much/too little, etc.

just to be clear, if the user has a `print` statement in transformed data block, it will go to stderr; otherwise, stdout.
but yes, agree, this is terrible.

Reply:
Hmm I guess we can go ahead and fix this streaming thing then.

I was looking at the generated cpp for a model:

```cpp
stan::model::model_base& new_model(
        stan::io::var_context& data_context,
        unsigned int seed,
	std::ostream* msg_stream) {
  stan_model* m = new stan_model(data_context, seed, msg_stream);
  return *m;
}
```

Then I went to look at the stan_model constructor in my generated code and the pstream__ argument is unused there. Where does this argument come in to play anyway?

Reply:
Or I guess the question is how does this argument end up changing where stuff goes anyway? Is this a problem with how some service function is not using loggers where it should be or something weird?

Reply:
I'd love to figure out what's going on here and then address it in a future PR.
for now,  can let thing go to stdout for now and just simplify return codes with this PR.